File: scraper/pushToDB.js
import fetch from 'node-fetch';
import dotenv from 'dotenv';

dotenv.config();
const { USERNAME, PASSWORD, URL } = process.env;

const feedId = process.argv[2];
const id = process.argv[3];

let systemContent = '';
process.stdin.setEncoding('utf8');

process.stdin.on('data', function(chunk) {
  systemContent += chunk;
});

process.stdin.on('end', async () => {
  try {
    // console.log(systemContent);

    if (!systemContent.trim()) {
      console.error('No content to push to db');
      process.exit(1);
    }

    const parsedContent = JSON.parse(systemContent);
    const { summary, tags, advertising } = parsedContent;

    const response = await fetch(URL + '/api/v1/news', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Basic ' + Buffer.from(USERNAME + ':' + PASSWORD).toString('base64')
      },
      body: JSON.stringify({
        feedId: feedId,
        originalFeedItemId: id,
        text: summary,
        tags: tags,
        advertising: advertising
      })
    });

    if (response.status != 200) {
      console.error('Call to news-api failed. ', response);
      process.exit(1);
    }

  } catch (error) {
    console.error(error);
    process.exit(1);
  }
});

File: scraper/push_to_db.py
import time
import requests
import config
from loguru import logger

def push_to_db(feed_id, item_id, parsed_content, max_retries=3):
    payload = {
        'feedId': feed_id,
        'originalFeedItemId': item_id,
        'text': parsed_content.get('summary'),
        'tags': parsed_content.get('tags'),
        'advertising': parsed_content.get('advertising')
    }

    for attempt in range(1, max_retries + 1):
        response = requests.post(
            f"{config.URL}/api/v1/news",
            json=payload,
            auth=(config.USERNAME, config.PASSWORD),
            headers={'Content-Type': 'application/json'}
        )
        try:
            response.raise_for_status()
            return
        except requests.HTTPError:
            if response.status_code >= 500 and attempt < max_retries:
                wait = attempt * 10
                logger.warning(f"Call to news-api failed (attempt {attempt}/{max_retries}). {response.status_code}: {response.text}. Retrying in {wait}s...")
                time.sleep(wait)
            else:
                raise RuntimeError(f"Call to news-api failed. {response.status_code}: {response.text}")
